config/config.go

```diff
@@ -986,20 +965,44 @@ type ConsensusConfig struct {
 	PeerQueryMaj23SleepDuration time.Duration `mapstructure:"peer-query-maj23-sleep-duration"`
 
 	DoubleSignCheckHeight int64 `mapstructure:"double-sign-check-height"`
+
+	// TODO: The following fields are all temporary overrides that should exist only
+	// for the duration of the v0.36 release. The below fields should be completely
+	// removed in the v0.37 release of Tendermint.
+
+	// UnsafeProposeTimeoutOverride provides an unsafe override of the Propose
+	// timeout consensus parameter. It configures how long the consensus engine
+	// will wait to receive a proposal block before prevoting nil.
+	UnsafeProposeTimeoutOverride *time.Duration `mapstructure:"unsafe-propose-timeout-override"`
```

Comment:
I'm not sure we necessarily these to be pointers. If they're set to zero, they're going to get default behaviour anyway, and whether that's done explicitly or via the config file don't we wind up with the same values?

Reply:
Hmm, the distinction I'm trying to make is between not set vs. 0. 
In the not-set case, we use the Consensus Params value for the corresponding parameter. However, it appears that, historically, 0 has been a valid value for these parameters, so making the distinction between 0 and not set would preserve this historical behavior.

Reply:
Hmm. It seems like if anyone had ever set those values to zero (or even "very small positive" values) consensus would not be working. On that basis, I feel like we could safely reuse zero as a "use the default" sign: Since the consensus params will default to the (given) defaults anyway, that would make the stock config work the same way.

This isn't a huge problem, it just seems a shame to add an additional degree of freedom unless we really need to. But if I'm missing something further, I don't think it's something to block for.

Reply:
In my research from ADR-74, I discovered that the timeout fields are rarely set to anything beyond the default. I feel very little worry about requiring these overrides to use positive values. Allowing these parameters to be nil adds a good deal of complexity that I'm not sure is worth if for the ability to set the parameters to 0, especially when setting them to `1 * time.Nanosecond` is nearly equivalent. 

Changed to not use `nil`s and only use these parameters when they are greater than 0. This also means we no longer rely on parser/deserializer behavior which feels much more future-proof against changes in the `toml` parser.